app/src/main/java/pl/droidcon/app/database/DatabaseManager.java

```diff
@@ -1,346 +1,147 @@
-        List<DataObserver> dataObservers = dataObserverMap.get(dataObserver.getType());
-        if (dataObservers == null) {
-            throw new IllegalStateException("Not found any registered observers for given type " + dataObserver.getType());
-        }
-        if (!dataObservers.contains(dataObserver)) {
-            throw new IllegalStateException("Observer not registered");
-        }
-        dataObservers.remove(dataObserver);
     }
 
-    public Observable<List<Session>> sessions(final SessionDay sessionDay) {
-        return RealmObservable.results(context, new Func1<Realm, RealmResults<RealmSession>>() {
-            @Override
-            public RealmResults<RealmSession> call(Realm realm) {
-                Log.d(TAG, "getting sessions from db for day=" + sessionDay + " and transforming to base models");
-                //its a hack
-                Date beginDate = sessionDay.when.toDate();
-                Date endOfDate = sessionDay.when.plusHours(23).toDate();
-                return realm
-                        .where(RealmSession.class)
-                        .between("date", beginDate, endOfDate)
-                        .findAll();
-            }
-        }).map(new Func1<RealmResults<RealmSession>, List<Session>>() {
-            @Override
-            public List<Session> call(RealmResults<RealmSession> realmSessions) {
-                List<Session> sessions = sessionMapper.fromDBList(realmSessions);
-                Collections.sort(sessions);
-                return sessions;
-            }
-        });
-    }
+    public Observable<Result<SessionEntity>> sessions(final SessionDay sessionDay) {
+        Date beginDate = sessionDay.when.toDate();
+        Date endOfDate = sessionDay.when.plusHours(23).toDate();
 
-    public Observable<List<Session>> sessions(final Collection<Integer> sessionIds) {
-        return RealmObservable.results(context, new Func1<Realm, RealmResults<RealmSession>>() {
-            @Override
-            public RealmResults<RealmSession> call(Realm realm) {
-                return realm.where(RealmSession.class)
-                        .findAll();
-            }
-        }).map(new Func1<RealmResults<RealmSession>, List<Session>>() {
-            @Override
-            public List<Session> call(RealmResults<RealmSession> realmSessions) {
-                List<RealmSession> rightSessions = new ArrayList<>();
-                for (RealmSession realmSession : realmSessions) {
-                    if (sessionIds.contains(realmSession.getId())) {
-                        rightSessions.add(realmSession);
-                    }
-                }
-                return sessionMapper.fromDBList(rightSessions);
-            }
-        });
+        return store
+                .select(SessionEntity.class)
+                .where(SessionEntity.DATE.between(beginDate, endOfDate))
+                .get().toSelfObservable();
```

Comment:
Co widze po zachowaniu `toSelfObservable`, ze kazdy INSERT powoduje wywolanie `onNext` w `AgendaFragment`.
Konczy sie to tym ze UI jest wolany ~1700 razy :) Sprobuje rozgryzc powod tego.